ui/app/src/components/shared/airlock/Airlock.tsx

```diff
@@ -0,0 +1,228 @@
+        ariaLabel: 'Creator of the airlock request',
+        minWidth: 150,
+        maxWidth: 200,
+        isResizable: true,
+        onRender: (request: AirlockRequest) => request.user?.name,
+        onColumnClick: reorderColumn
+      },
+      {
+        key: 'type',
+        name: 'Type',
+        ariaLabel: 'Whether the request is import or export',
+        minWidth: 70,
+        maxWidth: 100,
+        isResizable: true,
+        fieldName: 'requestType',
+        onColumnClick: reorderColumn
+      },
+      {
+        key: 'status',
+        name: 'Status',
+        ariaLabel: 'Status of the request',
+        minWidth: 70,
+        isResizable: true,
+        fieldName: 'status',
+        onColumnClick: reorderColumn
+      },
+      {
+        key: 'created',
+        name: 'Created',
+        ariaLabel: 'When the request was created',
+        minWidth: 120,
+        data: 'number',
+        isResizable: true,
+        fieldName: 'createdTime',
+        onRender: (request: AirlockRequest) => {
+          return <span>{ moment.unix(request.creationTime).format('DD/MM/YYYY') }</span>;
+        },
+        onColumnClick: reorderColumn
+      },
+      {
+        key: 'updated',
+        name: 'Updated',
+        ariaLabel: 'When the request was last updated',
+        minWidth: 120,
+        data: 'number',
+        isResizable: true,
+        isSorted: true,
+        fieldName: 'updatedWhen',
+        onRender: (request: AirlockRequest) => {
+          return <span>{ moment.unix(request.updatedWhen).fromNow() }</span>;
+        },
+        onColumnClick: reorderColumn
+      }
+    ];
+    setRequestColumns(columns);
+  }, []);
+
+  let requestsList;
+  switch (loadingState) {
+    case 'ok':
+      if (airlockRequests.length > 0) {
+        requestsList = (
+          <DetailsList
+            items={airlockRequests}
+            columns={requestColumns}
+            selectionMode={SelectionMode.none}
+            getKey={(item) => item.id}
+            onItemInvoked={(item) => navigate(item.id)}
+            className="tre-table-rows-align-centre"
+          />
+        );
+      } else {
+        requestsList = (
+          <div style={{textAlign: 'center', padding: '50px'}}>
+            <h4>No requests found</h4>
+            <small>Looks like there are no airlock requests yet. Create a new request to get started.</small>
+          </div>
+        )
+      }
+      break;
+    case 'error':
+      requestsList = (
+        <MessageBar
+          messageBarType={MessageBarType.error}
+          isMultiline={true}
+        >
+          <h3>Error fetching airlock requests</h3>
+          <p>There was an error fetching the airlock requests. Please see the browser console for details.</p>
+        </MessageBar>
+      ); break;
+    default:
+      requestsList = (
+        <div style={{ padding: '50px' }}>
+          <Spinner label="Loading airlock requests" ariaLive="assertive" labelPosition="top" size={SpinnerSize.large} />
```

Comment:
feel free to ignore but might be nice to use a `<Shimmer>` instead of a spinner, if we're loading a table of data?